client/core/state_transition.go

```diff
@@ -201,13 +224,33 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
 		// sufficient balance to make the transfer happen. The first
 		// balance transfer may never fail.
 		if vmerr == vm.ErrInsufficientBalance {
-			return nil, 0, false, vmerr
+			return nil, 0, common.Big0, false, vmerr
 		}
 	}
+
 	st.refundGas()
-	st.state.AddBalance(st.evm.Coinbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.gasPrice))
+	
+	computeFees := new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.gasPrice)
 
-	return ret, st.gasUsed(), vmerr != nil, err
+	// stabily fee enabled
+	if st.initialStabilityFee.Cmp(common.Big0) > 0 {
+		// calculate the final stability fee and refund the remaining value.
+		if st.gas > 0 {
```

Comment:
so what happens if no gas and has initial stability fee?

Reply:
nothing :) (no refund). merged the ifs